src/extension/background-script/events/persistPayments.js

```diff
@@ -1,16 +1,16 @@
 import db from "../db";
 
 const persistSuccessfullPayment = async (message, data) => {
-  const recipientName = data.origin.name;
-  const host = data.origin.host;
+  const host = data?.origin?.host || "";
+  const location = data?.origin?.location || "";
   const paymentResponse = data.response;
   const route = paymentResponse.data.route;
   const { total_amt, total_fees } = route;
 
   await db.payments.add({
-    host: host,
-    location: data.origin.location,
-    name: recipientName,
+    host,
+    location,
+    name: "", // we never know the "receiver" when using `confirmPayment`. To avoid setting a website as "receiver" we omit this.
```

Comment:
I don't understand this. 
and this is also different to how it currently is. we stored the `recipientName` which is `data.origin.name`;